### PR TITLE
port to Gtk3

### PR DESCRIPTION
**Pull request by uhoreg**
I'm not a Gtk expert, but from my quick testing, this seems to work.

I've also changed the ordering of the file picker dialogs to be consistent with each other (using the order preferred by the GNOME HIG)

### CODE REVIEW

File: gloc [hunk 1]
```diff
@@ -445,25 +445,25 @@ for (1..$n_threads) {
...
 ## Build main GUI
-my $mw = Gtk2::Window->new( 'toplevel' );
+my $mw = Gtk3::Window->new( 'toplevel' );
 $mw->signal_connect( 'delete_event' => \&clean_quit );
 $mw->set_default_size( 400, 500 );
 $mw->set_title( "$PROG_NAME v$VERSION" );
 
 ## Set up icon theme
-my $it = Gtk2::IconTheme->get_default();
+my $it = Gtk3::IconTheme::get_default();
```
Comment (jvolkening) [inline gloc]: Since `add_builtin_icon` is now a function, this line is no longer needed.

File: gloc [hunk 2]
```diff
@@ -1581,7 +1581,7 @@ sub get_embedded {
 
     for (keys %encoded) {
         $embedded->{$_} = do {
-            my $loader = Gtk2::Gdk::PixbufLoader->new();
+            my $loader = Gtk3::Gdk::PixbufLoader->new();
             $loader->write( decode_base64( $encoded{$_} ) );
```
Comment (jvolkening) [inline gloc]: From testing, it appears that a change is needed here to the format of the argument to `write()`. I suggest:
```
my $img_data = decode_base64($encoded{$_});
$loader->write( [unpack 'C*', $img_data] );
```

**Review by jvolkening — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good except for the two places indicated.

**Comment by uhoreg** [comment]
Thanks for the quick review.  I've made the changes as requested.

**Review by jvolkening — APPROVED** [review]

**Comment by jvolkening** [comment]
Thank you @uhoreg. I will add you as a contributor before merging this to master.

**Pull request merged**